[ci] current master branch macOS builds fail

`/usr/local/Cellar/gcc/8.2.0/lib/gcc/8/gcc/x86_64-apple-darwin17.7.0/8.2.0/include-fixed/stdio.h:78:10: fatal error: _stdio.h: No such file or directory`

The same issue: #1420.

Log:
```
-- The C compiler identification is GNU 8.2.0
-- The CXX compiler identification is GNU 8.2.0
-- Checking whether C compiler has -isysroot
-- Checking whether C compiler has -isysroot - yes
-- Checking whether C compiler supports OSX deployment target flag
-- Checking whether C compiler supports OSX deployment target flag - yes
-- Check for working C compiler: /usr/local/bin/gcc-8
-- Check for working C compiler: /usr/local/bin/gcc-8 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking whether CXX compiler has -isysroot
-- Checking whether CXX compiler has -isysroot - yes
-- Checking whether CXX compiler supports OSX deployment target flag
-- Checking whether CXX compiler supports OSX deployment target flag - yes
-- Check for working CXX compiler: /usr/local/bin/g++-8
-- Check for working CXX compiler: /usr/local/bin/g++-8 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenMP_C: -fopenmp  
-- Found OpenMP_CXX: -fopenmp  
-- Found OpenMP: TRUE   
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/travis/build/Microsoft/LightGBM/build
Scanning dependencies of target _lightgbm
[  3%] Building CXX object CMakeFiles/_lightgbm.dir/src/c_api.cpp.o
In file included from /usr/include/wchar.h:90,
                 from /usr/local/Cellar/gcc/8.2.0/include/c++/8.2.0/cwchar:44,
                 from /usr/local/Cellar/gcc/8.2.0/include/c++/8.2.0/bits/postypes.h:40,
                 from /usr/local/Cellar/gcc/8.2.0/include/c++/8.2.0/bits/char_traits.h:40,
                 from /usr/local/Cellar/gcc/8.2.0/include/c++/8.2.0/string:40,
                 from /usr/local/Cellar/gcc/8.2.0/include/c++/8.2.0/stdexcept:39,
                 from /Users/travis/build/Microsoft/LightGBM/include/LightGBM/utils/openmp_wrapper.h:7,
                 from /Users/travis/build/Microsoft/LightGBM/src/c_api.cpp:1:
/usr/local/Cellar/gcc/8.2.0/lib/gcc/8/gcc/x86_64-apple-darwin17.7.0/8.2.0/include-fixed/stdio.h:78:10: fatal error: _stdio.h: No such file or directory
 #include <_stdio.h>
          ^~~~~~~~~~
compilation terminated.
make[3]: *** [CMakeFiles/_lightgbm.dir/src/c_api.cpp.o] Error 1
make[2]: *** [CMakeFiles/_lightgbm.dir/all] Error 2
make[1]: *** [CMakeFiles/_lightgbm.dir/rule] Error 2
make: *** [_lightgbm] Error 2
```

Possible reason: ~gcc has been updated to 8.2.0~ disproved by #1559,

![image](https://user-images.githubusercontent.com/25141164/43490855-f7943ea4-952a-11e8-9935-9d00546003b5.png)

--- a/.travis/setup.sh
+++ b/.travis/setup.sh
@@ -1,6 +1,7 @@
 #!/bin/bash
 
 if [[ $TRAVIS_OS_NAME == "osx" ]]; then
+    sudo softwareupdate -i "Command Line Tools (macOS High Sierra version 10.13) for Xcode-9.3"  # fix "fatal error: _stdio.h: No such file or directory"
     rm '/usr/local/include/c++'
 #    brew cask uninstall oclint  #  reserve variant to deal with conflict link
     if [[ $TASK == "mpi" ]]; then

